### DISCUSSION TITLE
I am currently using ethers in my project, can I use it with wagmi?

### DISCUSSION THREAD

**Discussion by ghost** [wevm/wagmi#2646] [Q&A]
Due to Web3Modalv2 upgrade, I had to make changes to my project. It uses wagmi and viem in w3modalv2 but my project is ethersjs heavy. So removing ethersjs will completely ruin the project. 

## Can I use ethersjs with wagmi?

**Comment by tmm** [comment]
You can continue to use ethers if you want, although wagmi internally is powered by viem. To get the most benefit, you should phase out ethers and switch to viem entirely as it has a smaller bundle size, is significantly faster (think encoding/decoding), better type safety, etc.

**Reply by ghost** [reply]
how to update wagmi page when network changes?

https://docs.cloud.coinbase.com/wallet-sdk/v2.0.0/docs/web3modal
I've done it before by looking at this documentation. Wagmi and web3modal documentation is very messy and my project is bogged down

**Reply by ghost** [reply]
I want to make an api call when the wallet is connected is there any example
